### PR TITLE
*: refactor interface to make lint happy

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <chris@lonng.org>

### What problem does this PR solve? 

Refactor `Eval/EvalContext/InjectContext` to make code more idiom.

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

N/A

Related changes

N/A

### CODE REVIEW

File: code/expr_rewriter.go [hunk 1]
```diff
@@ -111,11 +111,12 @@ func (r *Rewriter) rewriteInjectContext(call *ast.CallExpr) (bool, ast.Stmt, err
 	if len(call.Args) != 3 {
 		return false, nil, fmt.Errorf("failpoint.InjectContext: expect 3 arguments but got %v", len(call.Args))
 	}
-	fpname, ok := call.Args[0].(*ast.BasicLit)
+
+	ctxname, ok := call.Args[0].(*ast.Ident)
 	if !ok {
 		return false, nil, fmt.Errorf("failpoint.InjectContext: first argument expect string literal but got %T", call.Args[0])
```
Comment (kennytm) [inline code/expr_rewriter.go]: Update this error message.

File: code/expr_rewriter.go [hunk 2]
```diff
@@ -111,11 +111,12 @@ func (r *Rewriter) rewriteInjectContext(call *ast.CallExpr) (bool, ast.Stmt, err
...
 	}
-	ctxname, ok := call.Args[1].(*ast.Ident)
+	fpname, ok := call.Args[1].(*ast.BasicLit)
 	if !ok {
 		return false, nil, fmt.Errorf("failpoint.InjectContext: second argument expect string literal but got %T", call.Args[0])
```
Comment (kennytm) [inline code/expr_rewriter.go]: (ditto)

**Comment by lonng** [comment]
PTAL @amyangfei

**Comment by lonng** [comment]
> also update the usage of `InjectContext` in README
> REST LGTM

OK, I will do it after https://github.com/pingcap/failpoint/pull/10 merged

**Comment by lonng** [comment]
PTAL again  @kennytm @amyangfei

**Pull request merged**